src/compiler/scanner.ts

```diff
@@ -2066,6 +2063,32 @@ namespace ts {
             }
         }
 
+        function reScanInvalidIdentifier(): SyntaxKind {
+            Debug.assert(token === SyntaxKind.Unknown, "'reScanInvalidIdentifier' should only be called when the current token is 'SyntaxKind.Unknown'.");
+            pos = tokenPos = startPos;
+            tokenFlags = 0;
+            const ch = codePointAt(text, pos);
+            const identifierKind = scanIdentifier(ch, ScriptTarget.ESNext);
+            if (identifierKind) {
+                return token = identifierKind;
+            }
+            pos += charSize(ch);
+            return token = SyntaxKind.Unknown;

```

Comment:
Do you need to reassign token here? From what I can tell:
- You assert `token` is `Unknown` at the top.
- `scanIdentifier` (by way of `getIdentifierToken`) only sets `token` if the scan matched a valid keyword or identifier.

So it should still be `Unknown` here.

Reply:
You’re right, copy-paste oversight. That said, I think this is certainly easier to read/understand (although a comment could be just as good).